src/lib.rs

```diff
@@ -85,8 +85,11 @@ extern crate alloc;
 #[cfg(feature = "std")]
 extern crate std as alloc;
 
-#[cfg_attr(feature = "std", path = "std.rs")]
-#[cfg_attr(not(feature = "std"), path = "no_std.rs")]
+#[cfg_attr(any(feature = "std", feature = "critical-section"), path = "std.rs")]
+#[cfg_attr(
+    not(any(feature = "std", feature = "critical-section")),
+    path = "no_std.rs"
+)]
```

Comment:
hmm.. I'm a bit confused. I thought `critical-section` is for no_std but here it seems we're `std` module for both? :thinking:

Reply:
Maybe I should rename the `std` and `no_std` modules to something else, now that they don't correspond to `std` and `no_std` more. Maybe `intrusive` vs `slab`, since that's what the implementation is based on.

Reply:
> Maybe `intrusive` vs `slab`, since that's what the implementation is based on.

Sounds good to me.